Remove associated trips when removing calendar_dates

landonreed opened the pull request:
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR fixes #339 by checking if a service_id exists in the merge feed result and skipping trips if no ref is found. It also fixes some embedded issues with checking the correct variable (`valueToWrite` vs `value`).

Note: there is some clumsiness around the serviceIdMissing check and the subsequent check that it is not missing. I'm open to suggestions about improving this: https://github.com/ibi-group/datatools-server/pull/340/files#diff-b601a83ada6de294158d19894ceefe57R872-R883

landonreed committed:
test(feed-merge): add condition to MTC feed merge test

re #339

landonreed committed:
fix(feed-merge): remove trips that ref service_ids skipped in merge

fix #339

landonreed commented:
I think this still needs some work/an additional test. Re-assigning to myself.

landonreed committed:
refactor(feed-merge): correctly handle removal of trips for missing service_id

binh-dam-ibigroup reviewed (approved):
Conditional approval on passing the CI tests.

landonreed committed:
ci(travis): only print errors during package to limit log output

re #342

landonreed committed:
refactor(travis): fix mvn package command (run quiet)

re #342

evansiroky committed:
fix: update to latest gtfs-lib version

evansiroky committed:
build: add geotools dependency

this geotools dependency includes coordinate reference libraries and was removed in the latest gtfs-lib release in https://github.com/conveyal/gtfs-lib/pull/290

landonreed committed:
refactor(feed-merge): apply service_id checks only during service period merge
